src/core/config/reader.rs

```diff
@@ -131,16 +129,16 @@ impl ConfigReader {
                     let meta = self.proto_reader.fetch(link.src.as_str()).await?;
 
                     for m in meta {
-                        config_module.extensions.add_proto(m);
+                        extensions.add_proto(m);
                     }
                 }
             }
         }
 
         // Recreating the ConfigModule in order to recompute the values of
         // `input_types`, `output_types` and `interface_types`
-        let mut final_config_module = ConfigModule::from(config_module.config);
-        final_config_module.extensions = config_module.extensions;
+        let mut final_config_module = ConfigModule::from(config_module.config().clone());
+        final_config_module.with_extensions(extensions);
```

Comment:
This is just a bit awkward. I think the main problem is that we are modifying the `config` above when there is a `LinkType::Config` and then we use `merge_right` on `ConfigModule`. 

I think a better approach would be to pull out the config from config_module modify it wherever required and then recreate the ConfigModule. We should not use the `merge_right` feature in ConfigModule, it's derived and is logically incorrect.